src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx

```diff
@@ -11,34 +11,65 @@ import {
 } from "@chakra-ui/react"
 import { Button, ModalCloseButton } from "@opengovsg/design-system-react"
 import _ from "lodash"
-import Select, { PropsValue, useStateManager } from "react-select"
+import { useEffect } from "react"
+import { Controller, FormProvider, useForm } from "react-hook-form"
+import { useParams } from "react-router-dom"
+import Select from "react-select"
+
+import { useUpdateReviewRequest } from "hooks/reviewHooks/useUpdateReviewRequest"
+
+import { getAxiosErrorMessage } from "utils/axios"
 
 import { User } from "types/reviewRequest"
+import { useErrorToast, useSuccessToast } from "utils"
 
 export interface ManageReviewerModalProps extends Omit<ModalProps, "children"> {
   selectedAdmins: User[]
   admins: User[]
 }
 
-// NOTE: This is with reference to
-// https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/types.ts
-// The initial type is PropsValue, which is either a single value or a multi-value.
-// We need to narrow the type ourself to prove to the TS compiler that this is a list.
-const isList = (admins: PropsValue<unknown>): admins is unknown[] => {
-  // NOTE: Required as SingleValue can be null
-  if (!admins) {
-    return false
-  }
-
-  return !!(admins as User[]).length
-}
-
 export const ManageReviewerModal = ({
   selectedAdmins,
   admins,
   ...props
 }: ManageReviewerModalProps): JSX.Element => {
   const { onClose } = props
+  const { siteName, reviewId } = useParams<{
+    siteName: string
+    reviewId: string
+  }>()
+  const prNumber = parseInt(reviewId, 10)
+  const successToast = useSuccessToast()
+  const errorToast = useErrorToast()
+
+  const {
+    mutateAsync: updateReviewRequest,
+    isLoading: isUpdateReviewRequestLoading,
+    isError: isUpdateReviewRequestError,
+    error: updateReviewRequestError,
+    isSuccess: isUpdateReviewRequestSuccess,
+  } = useUpdateReviewRequest(siteName, prNumber)
+
+  const methods = useForm<{ reviewers: User[] }>()
+
+  const onSubmit = methods.handleSubmit(async (data) => {
+    await updateReviewRequest(data)
+    onClose()
+  })
+
+  useEffect(() => {
+    if (isUpdateReviewRequestSuccess) {
+      successToast({ description: "Your review request has been updated!" })
+    }
+  }, [isUpdateReviewRequestSuccess, successToast])
+
+  useEffect(() => {
+    if (isUpdateReviewRequestError) {
+      errorToast({
+        description: getAxiosErrorMessage(updateReviewRequestError),
+      })
+    }
+  }, [errorToast, isUpdateReviewRequestError, updateReviewRequestError])
```

Comment:
Nit: should this be placed in the hook itself instead, via `onError`/`onSuccess`? It doesn't seem to require anything in particular from the component, and the error/success toast should be the same regardless of where it's called

Reply:
no real opinion here - i've just always been placing it inside the component cos i think that's where devs expect to see the toast + gives the component/page the ability to give a better error message in the event that it's a 500.

not too attached though; wdyt?

Reply:
No particularly strong feelings either way also - I'm fine with leaving it as is!

Reply:
will take path of least resistance